server/routes/question.py

```diff
@@ -0,0 +1,35 @@
+from fastapi import APIRouter
+from fastapi.responses import FileResponse
+from server.dto import GPTAudioRequest
+from fastapi.responses import StreamingResponse
+from typing import List
+
+from dotenv import load_dotenv
+import os
+import json
+from src.app import main
+router = APIRouter()
+
+def file_generator(file_paths: List[str]):
+    for file_path in file_paths:
+        with open(file_path, "rb") as file:
+            yield file.read()
+
+@router.post("/GPTAudio/")
+async def generate_audio(req: GPTAudioRequest) -> StreamingResponse:
+    text_path = os.getenv('TEXT_PATH')
+    with open(f"{text_path}", "w") as f:
+        f.write(req.question)
+    load_dotenv(dotenv_path=".env", override=True)
+    main()
+
+    # load audio file paths from output.json
+    output_json_folder = os.getenv('OUTPUT_DIR')
+    with open(f"{output_json_folder}/output.json", 'r') as f:
+        dict = json.load(f)
+    print(dict)
+    audio_paths = list()
+    for item in dict["content"]:
+        audio_paths.append(item["audio_file_path"])
+
+    return StreamingResponse(content=file_generator(audio_paths), media_type="audio/mpeg")
```

Comment:
Will this wait for all audios to finish or will it return the response when all files are ready, because it looks to me you would have to append all the audios to audio_paths first then stream the through the audio paths, what if in dict['content'] the item['audio_file_path'] is not ready what happens, will it just skip the audio and continue. then this will result in skipped content, or will this throw an error, then what happens? Maybe we can think of another way, maybe try searching for async await or yield or maybe something else? Good luck!😉